paasta_tools/kubernetes_tools.py

```diff
@@ -2614,9 +2625,22 @@ def sanitize_for_config_hash(
 
         return ahash
 
-    def get_termination_grace_period(self) -> Optional[int]:
-        return self.config_dict.get("lifecycle", KubeLifecycleDict({})).get(
-            "termination_grace_period_seconds"
+    def get_termination_grace_period(
+        self, service_namespace_config: ServiceNamespaceConfig
+    ) -> Optional[int]:
+        """Return the number of seconds that kubernetes should wait for pre-stop hooks to finish (or for the main
+        process to exit after signaling) before forcefully terminating the pod.
+
+        Defaults to a value long enough to allow the standard hadown draining to finish.
+        """
+
+        return self.get_lifecycle_dict().get(
+            "termination_grace_period_seconds",
+            (
+                self.get_hacheck_prestop_sleep_seconds() + 1
+                if service_namespace_config.is_in_smartstack()
+                else None
+            ),
```

Comment:
I considered adding another conditional for whether `pre_stop_drain_seconds` is even specified in yelpsoa-configs (returning None if not) to avoid the big bounce for routable pods, but decided that would be too much complexity to avoid a small one-time pain.

I'm tempted to apply the same logic to the `is_in_smartstack` conditional, i.e. remove that conditional (and the extra argument) and just default to `self.get_hacheck_prestop_sleep_seconds() + 1` regardless. This would mean we'd change the default from `None` (which is really 30 seconds, since that's the default for k8s podSpec) to 32 seconds. Probably not a big deal in practice, and again just means we'd have a big bounce (of all pods, mesh-registered and not)

Reply:
either way is fine imo